### ISSUE TITLE
Callable correlation support

### ISSUE THREAD

**Issue by jmkim** [dask/dask#6550]
https://docs.dask.org/en/latest/dataframe-api.html#dask.dataframe.DataFrame.corr

Please support `callable` option for `DataFrame.corr`.

When I tried to run the callable function, the error shown as:
```
NotImplementedError: Only Pearson correlation has been implemented
```

Thank you!

**Comment by mrocklin** [comment]
I agree that this would be useful, but unfortunately it's probably hard to automatically parallelize an arbitrary callable function.  I'm not sure how this would be possible.

**Comment by TomAugspurger** [comment]
We can force the parallelization issues onto the user by requiring the callable to accept and return dask arrays. That might not help with people naively porting pandas to dask, but would at least give the option.
